lambdaclass/mirra_backend
# [GH-736] - Snapshots

## Motivation

Have different versions of our configurations and be able to switch back and forth between them. 

Closes #736 

## Summary of changes

- [Add MVC for version control](https://github.com/lambdaclass/mirra_backend/pull/806/commits/48f52f2301da1a5f626774b1a19b9834a2603f5b)
- [Add needed logic and an endpoint to retrieve the game's config by version](https://github.com/lambdaclass/mirra_backend/pull/806/commits/e145812123c582aabc588e100f33d0d6a9df1cea)
- [Add version id to map configurations](https://github.com/lambdaclass/mirra_backend/pull/806/commits/37c77977b641c1b438a1f41adff3ab56d293dffd)
- [Add logic to set a version as current and an endpoint to retrieve the current configuration asociated to that version](https://github.com/lambdaclass/mirra_backend/pull/806/commits/64dd68bbca80394ecd83b5e280530398e33179a1)
- [Add version id to game configuration](https://github.com/lambdaclass/mirra_backend/pull/806/commits/d1e706a4bb43d47030c30a841a694bc41f7385d8)
- [Removal of all current config getters from arena configuration and use the endpoint created to get the current configuration](https://github.com/lambdaclass/mirra_backend/pull/806/commits/d23a499ab67145e56d54655d2b6deeef153eabda)
- [Add version selector to all forms](https://github.com/lambdaclass/mirra_backend/pull/806/commits/70dc1e5980676a6b3e61ac0d2d9a8d66686c524d)
- [Display version on skill's show](https://github.com/lambdaclass/mirra_backend/pull/806/commits/0cb1b5938749c5726b5eb9a3731c1575490248ca)
- [Display version on character's show](https://github.com/lambdaclass/mirra_backend/pull/806/commits/52dd5842e104f287d466f9ea14b29b723d76a10c)
- [Display version on consumable items's show](https://github.com/lambdaclass/mirra_backend/pull/806/commits/c53733f9d48d751d4790a8f0a9d27affb104b28f)
- [Display version on map configuration's show](https://github.com/lambdaclass/mirra_backend/pull/806/commits/4452f7d809de822dc61817e1ebef12e6afd63b82)
- [Display version on game configuration's show](https://github.com/lambdaclass/mirra_backend/pull/806/commits/9aeb881050a20d48cd66c2c46234700dedf4552f)
- [Add has_one and has_many relationship to version schema](https://github.com/lambdaclass/mirra_backend/pull/806/commits/d6597454eaec8dae75711b144e9456f5450c03db)
- [Use the has one and many relationships to get the data from a current version](https://github.com/lambdaclass/mirra_backend/pull/806/commits/10b52bbe5c3d159ff8ddb0d6a4876eb116a55cf7)

## How to test it?

1. In local development using seeds:
    1. Run `make start` 
    2. Do a curl to `curl -X GET  http://localhost:4001/curse/configuration/current`
    3. You'll receive the current configuration for arena
    4. Then, play a game and check that everything works well
    
2. In local development using seeds and creating a new version
    - Go to `localhost:4100/versions` and create a new version
    - Now you have to create characters, map configurations, game configurations, etc that belongs to this version (Be careful, if you miss one of them, the game will not work)
    - Go to `localhost:4100/versions` and set the new version as current
    - Do a curl to `curl -X GET  http://localhost:4001/curse/configuration/current` and check that the configuration is different
    - Play a game and check that everything works well and uses the new configuration version

3. In live the configurator lets follow the same steps as option 2 in local development but using the live URL

## Checklist
